libs/dh/wholesale/feature-start/src/lib/dh-wholesale-start.component.ts

```diff
@@ -38,51 +62,101 @@ import { CommonModule } from '@angular/common';
   providers: [DhWholesaleBatchDataAccessApiStore],
   standalone: true,
   imports: [
-    WattButtonModule,
-    WattDatepickerModule,
-    WattFormFieldModule,
-    TranslocoModule,
+    CommonModule,
     DhFeatureFlagDirectiveModule,
+    PushModule,
     ReactiveFormsModule,
-    CommonModule,
+    TranslocoModule,
+    WattButtonModule,
+    WattDatepickerModule,
     WattDropdownModule,
+    WattFormFieldModule,
   ],
 })
-export class DhWholesaleStartComponent {
-  constructor(private store: DhWholesaleBatchDataAccessApiStore) {}
-
-  optionsGridAreas: WattDropdownOption[] = Object.keys({
-    805: '805',
-    806: '806',
-    512: '512',
-    533: '533',
-    543: '543',
-    584: '584',
-    351: '351',
-  }).map((key) => ({
-    displayValue: key,
-    value: key,
+export class DhWholesaleStartComponent implements OnInit, OnDestroy {
+  private store = inject(DhWholesaleBatchDataAccessApiStore);
+  private toast = inject(WattToastService);
+  private transloco = inject(TranslocoService);
+  private router = inject(Router);
+
+  private destroy$ = new Subject<void>();
+
+  loadingCreatingBatch$ = this.store.loadingCreatingBatch$;
+
+  createBatchForm = new FormGroup<CreateBatchFormValues>({
+    gridAreas: new FormControl(null, { validators: Validators.required }),
+    dateRange: new FormControl(null, {
+      validators: WattRangeValidators.required(),
+    }),
+  });
+
+  optionsGridAreas: WattDropdownOption[] = [
+    '351',
+    '512',
+    '533',
+    '543',
+    '584',
+    '805',
+    '806',
+  ].map((gridAreaCode) => ({
+    displayValue: gridAreaCode,
+    value: gridAreaCode,
   }));
 
-  formControlGridArea = new FormControl<string[] | null>(
-    null,
-    Validators.required
-  );
+  ngOnInit(): void {
+    // Close toast on navigation
+    this.router.events
+      .pipe(first((event) => event instanceof NavigationEnd))
+      .subscribe(() => {
+        this.toast.dismiss();
+      });
+
+    this.store.creatingBatchSuccessTrigger$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(() => this.onBatchCreatedSuccess());
+
+    this.store.creatingBatchErrorTrigger$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(() => this.onBatchCreatedError());
+  }
 
-  formControlRange = new FormControl<{ start: string; end: string } | null>(
-    null,
-    [WattRangeValidators.required()]
-  );
```

Comment:
`dh-wholesale-start.component.spec.ts` is failing because it doesn't what `formControlRange` & `formControlGridArea` is.

Is the test no longer relevant or should it be refactored or should the control definitions be added back?

![Capture](https://user-images.githubusercontent.com/2743537/211287124-97364fd6-8884-4b6e-9fe5-99b0a1203a09.PNG)

![Capture1](https://user-images.githubusercontent.com/2743537/211287517-9e28a691-778c-492a-9f08-32808a98c407.PNG)